test/unit/modules/multisignatures.js

```diff
@@ -0,0 +1,443 @@
+						});
+
+						describe('when account exists', function () {
+
+							describe('when multisignature already contains tx.signature', function () {
+
+								it('should call callback with error = "Signature already exists"');
+							});
+
+							describe('for every account.multisignatures', function () {
+
+								it('should call library.logic.transaction.verifySignature');
+
+								it('should call library.logic.transaction.verifySignature with multisignature');
+
+								it('should call library.logic.transaction.verifySignature with account.multisignatures');
+
+								it('should call library.logic.transaction.verifySignature with tx.signature');
+
+								describe('when library.logic.transaction.verifySignature throws', function () {
+
+									it('should call library.logger.error');
+
+									it('should call library.logger.error with error stack');
+
+									it('should call callback with error = "Failed to verify signature"');
+								});
+
+								describe('when library.logic.transaction.verifySignature returns false', function () {
+
+									it('should call callback with error = "Failed to verify signature"');
+								});
+
+								describe('when library.logic.transaction.verifySignature returns true', function () {
+
+									continueSignatureProcessing();
+								});
+							});
+						});
+					});
+				});
+
+				describe('when multisignature transaction type = transactionTypes.MULTI', function () {
+
+					describe('when multisignature is already signed', function () {
+
+						it('should call callback with error = "Permission to sign transaction denied"');
+					});
+
+					describe('when multisignature already contains tx.signature', function () {
+
+						it('should call callback with error = "Permission to sign transaction denied"');
+					});
+
+					describe('for every multisignature keysgroup member', function () {
+
+						it('should call library.logic.transaction.verifySignature');
+
+						it('should call library.logic.transaction.verifySignature with multisignature');
+
+						it('should call library.logic.transaction.verifySignature with keysgroup member');
+
+						it('should call library.logic.transaction.verifySignature with tx.signature');
+
+						describe('when library.logic.transaction.verifySignature throws', function () {
+
+							it('should call library.logger.error');
+
+							it('should call library.logger.error with error stack');
+
+							it('should call callback with error = "Failed to verify signature"');
+						});
+
+						describe('when library.logic.transaction.verifySignature returns false', function () {
+
+							it('should call callback with error = "Failed to verify signature"');
+						});
+
+						describe('when library.logic.transaction.verifySignature returns true', function () {
+
+							continueSignatureProcessing();
+						});
+					});
+				});
+			});
+		});
+
+		function continueSignatureProcessing () {
```

Comment:
Please, move this function to the beginning or at least before using it.

Reply:
I prefer to keep it after - from the very beginning of reading the test you can easily follow the test path.

Reply:
agree with this.